databuilder/models/user.py

```diff
@@ -0,0 +1,163 @@
+        # type: (...) -> None
+        """
+        This class models user node for Amundsen people.
+
+        :param first_name:
+        :param last_name:
+        :param name:
+        :param email:
+        :param github_username:
+        :param team_name:
+        :param employee_type:
+        :param manager_email:
+        :param is_active:
+        :param updated_at: everytime we update the node, we will push the timestamp.
+                           then we will have a cron job to update the ex-employee nodes based on
+                           the case if this timestamp hasn't been updated for two weeks.
+        """
+        self.first_name = first_name
+        self.last_name = last_name
+        self.name = name
+        self.email = email
+        self.github_username = github_username
+        # todo: team will be a separate node once Amundsen People supports team
+        self.team_name = team_name
+        self.manager_email = manager_email
+        self.employee_type = employee_type
+        # this attr not available in team service, either update team service, update with FE
+        self.slack_id = slack_id
+        self.is_active = is_active
+        self.updated_at = updated_at
+
+        self._node_iter = iter(self.create_nodes())
+        self._rel_iter = iter(self.create_relation())
+
+    def create_next_node(self):
+        # type: (...) -> Union[Dict[str, Any], None]
+        # return the string representation of the data
+        try:
+            return next(self._node_iter)
+        except StopIteration:
+            return None
+
+    def create_next_relation(self):
+        # type: () -> Union[Dict[str, Any], None]
+        """
+        :return:
+        """
+        try:
+            return next(self._rel_iter)
+        except StopIteration:
+            return None
+
+    @classmethod
+    def get_user_model_key(cls,
+                           email=None):
+        # type: (...) -> str
+        if not email:
+            return ''
+        return User.USER_NODE_KEY_FORMAT.format(email=email)
+
+    def create_nodes(self):
+        # type: () -> List[Dict[str, Any]]
+        """
+        Create a list of Neo4j node records
+        :return:
+        """
+        result_node = {
+            NODE_KEY: User.get_user_model_key(email=self.email),
+            NODE_LABEL: User.USER_NODE_LABEL,
+            User.USER_NODE_EMAIL: self.email,
+            User.USER_NODE_IS_ACTIVE: self.is_active,
+        }
+
+        if self.first_name:
+            result_node[User.USER_NODE_FIRST_NAME] = self.first_name
+        if self.last_name:
+            result_node[User.USER_NODE_LAST_NAME] = self.last_name
+        if self.name:
+            result_node[User.USER_NODE_FULL_NAME] = self.name
+        if self.github_username:
+            result_node[User.USER_NODE_GITHUB_NAME] = self.github_username
+        if self.team_name:
+            result_node[User.USER_NODE_TEAM] = self.team_name
+        if self.manager_email:
```

Comment:
Wondering if we need this as we have manager relation relationship as we now could have info in two places.

Reply:
sync offline, given we now define the manager relationship, we don't need the manager info in the user attribute any more.